sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -512,6 +512,9 @@ class Analyzer(
 
       // A special case for Generate, because the output of Generate should not be resolved by
       // ResolveReferences. Attributes in the output will be resolved by ResolveGenerate.
+      case g @ Generate(generator, _, _, _, _, _)
+        if !g.resolved && generator.resolved => g
+
       case g @ Generate(generator, join, outer, qualifier, output, child)
```

Comment:
@davies Hi Davis, I also was thinking about it. I felt its probably safer to handle the generate plan by these two cases and not fall through the last case like we do for this defect. 
@cloud-fan What do you think ?